jobserver/views/projects.py

```diff
@@ -161,21 +166,26 @@ def get_outputs(self, workspaces):
 
         return Snapshot.objects.filter(pk__in=snapshot_pks).order_by("-published_at")
 
-    def get_repos(self, repo_urls):
-        for url in repo_urls:
+    def iter_repos(self, repo_urls):
+        def get_repo(url):
             f = furl(url)
 
             try:
                 is_private = get_repo_is_private(*f.path.segments)
             except requests.HTTPError:
                 is_private = None
 
-            yield {
+            return {
                 "name": f.path.segments[1],
                 "url": url,
                 "is_private": is_private,
             }
 
+        with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
I'd think this might be better as a global instance. Otherwise it will have to create the all the threads every request.

I would also recommend setting `thread_name_prefix="get_repo"` or similar, as when you have threading issues in the future, you'll wish you had.

It might be a good idea to set `max_workers` as well, perhaps, but it defaults to `min(32, cpu + 4)`, as long as its not unbounded I think thats fine.

Reply:
It also applies the max_workers across requests. Currently, it could be 32 threads per concurrent request, in theory!

Reply:
Fixed and rebased in.